src/lib/elementary/meson.build

```diff
@@ -262,16 +262,15 @@ endforeach
 
 eolian_include_directories += ['-I', meson.current_source_dir()]
 
-if sys_windows
-  embed_script = find_program('config_embed.bat')
-else
-  embed_script = find_program('config_embed')
-endif
+embed_config_exe = executable('embed_config_exe', files('../../bin/elementary/embed_config.c'), dependencies: getopt)
+meson.override_find_program('embed_config', embed_config_exe)
+
+embed_script_run = find_program('embed_config', required: false)
 
-embed_config = custom_target('create_embedded_default_config',
```

Comment:
Is there any reason to change this name?

Reply:
only thing came to my mind is that it is not a _script_ but an executable? I dont know

Reply:
welp...I think at least the `embed_config` should stay with the same name, but don't have an opinion about the script

Reply:
Actually this is not the executable, this is the "inline-header" generator, for a "config.h-like" header. See:

```meson
  output: 'elm_default_config.x',
  command: [embed_script, '@INPUT@', '@OUTPUT@']
```

So it uses the `embed_script_run` (which is the executable `embed_config_exe`) to generate the header (just like those `configuration_data` + `configure` to generate config.h, but using our executable to generate it instead of a sed-like internal procedure from meson). See the `.in` file used as input?

But this seems a bit overcomplicated. I don't see a reason (looking quickly) to override the `find_program` for `embed_config` (that `meson.override_find_program(a, b)` means "when you use `find_program(program name)` with `a` as program name, use the target `b` instead of anything it could find). Maybe `custom_target` doesn't accept meson targets in `command`?

Reply:
Ok, it actually can (see the `command` kwarg in [Meson docs about `custom_target`s](https://mesonbuild.com/Reference-manual.html#custom_target)), so this whole code could've be...

Reply:
```meson
embed_config = executable('embed_config_exe',
    files('../../bin/elementary/embed_config.c'),
    dependencies: getopt)

elementary_default_config = custom_target('create_embedded_default_config',
  input: join_paths(meson.source_root(), 'data', 'elementary', 'config', 'standard', 'base.src.in'),
  output: 'elm_default_config.x',
  command: [embed_script_run, '-o', '@OUTPUT@', '@INPUT@'],
)

**jumps to line 1026**

elementary_lib = library('elementary',
    elementary_src, pub_eo_file_target, priv_eo_file_target, c, elementary_default_config,
    dependencies: elementary_pub_deps + elementary_deps + elementary_ext_deps,
    include_directories : config_dir + [include_directories('.')] + [include_directories(join_paths('..', '..', '..'))],
    install: true,
    c_args : [elm_package_c_args, '-DELM_BUILD'],
    version : meson.project_version()
)
```

Reply:
To be honest, IMO `embed_config` should be generated in `src/bin/elementary`, since there's where the source actually is, and not here.

Reply:
> To be honest, IMO `embed_config` should be generated in `src/bin/elementary`, since there's where the source actually is, and not here.

The generator is from `src/bin/elementary` but the sources it is using are from `src/lib/elementary`, the place seems to be right to me.

Reply:
Oh, ok then.